`blayout= col_major` incorrectly emitted to `BLayout::RowMajor`, leading to wrong matmul kernel code

## Problem

At today's latest commit https://github.com/zhangstevenunity/PTOAS/commit/ca7bcc852ff8ca01d418d882826951d3fd1235e2, `ptoas` EmitC generates wrong `BLayout` for cube tiles. `blayout= col_major` all become `BLayout::RowMajor`, and `slayout` all become `NoneBox` regardless of its value.  Earlier commit https://github.com/zhangstevenunity/PTOAS/commit/6a8bcdbad7a0587c1d34b035617d36f1ee9c0cfa was working correctly.

For example this part of IR:

```mlir
      %19 = pto.alloc_tile : !pto.tile_buf<loc=mat, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=512, pad=0>
      %20 = pto.alloc_tile : !pto.tile_buf<loc=mat, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=512, pad=0>
      %21 = pto.alloc_tile : !pto.tile_buf<loc=left, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=row_major, slayout=row_major, fractal=512, pad=0>
      %22 = pto.alloc_tile : !pto.tile_buf<loc=right, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=row_major, slayout=col_major, fractal=512, pad=0>
      %23 = pto.alloc_tile : !pto.tile_buf<loc=acc, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=1024, pad=0>
```

incorrectly becomes:

```cpp
  Tile<TileType::Mat, float, 128, 128, BLayout::RowMajor, 128, 128, SLayout::NoneBox, 512, PadValue::Null> v25;
  TASSIGN(v25, v15);
  Tile<TileType::Mat, float, 128, 128, BLayout::RowMajor, 128, 128, SLayout::NoneBox, 512, PadValue::Null> v26;
  TASSIGN(v26, v16);
  Tile<TileType::Left, float, 128, 128, BLayout::RowMajor, 128, 128, SLayout::NoneBox, 512, PadValue::Null> v27;
  TASSIGN(v27, v16);
  Tile<TileType::Right, float, 128, 128, BLayout::RowMajor, 128, 128, SLayout::NoneBox, 512, PadValue::Null> v28;
  TASSIGN(v28, v16);
  Tile<TileType::Acc, float, 128, 128, BLayout::RowMajor, 128, 128, SLayout::NoneBox, 1024, PadValue::Null> v29;
  TASSIGN(v29, v16);
```

The correct version should be:

```cpp
  Tile<TileType::Mat, float, 128, 128, BLayout::ColMajor, 128, 128, SLayout::RowMajor, 512, PadValue::Null> v25;
  TASSIGN(v25, v15);
  Tile<TileType::Mat, float, 128, 128, BLayout::ColMajor, 128, 128, SLayout::RowMajor, 512, PadValue::Null> v26;
  TASSIGN(v26, v16);
  Tile<TileType::Left, float, 128, 128, BLayout::RowMajor, 128, 128, SLayout::RowMajor, 512, PadValue::Null> v27;
  TASSIGN(v27, v16);
  Tile<TileType::Right, float, 128, 128, BLayout::RowMajor, 128, 128, SLayout::ColMajor, 512, PadValue::Null> v28;
  TASSIGN(v28, v16);
  Tile<TileType::Acc, float, 128, 128, BLayout::ColMajor, 128, 128, SLayout::RowMajor, 1024, PadValue::Null> v29;
  TASSIGN(v29, v16);
```

## Reproduce

Run `ptoas matmul.pto -o matmul.cpp` on this IR:

```mlir
module attributes {"pto.device-spec" = "Ascend910B1"} {
  func.func @RunTMATMULSplitK(%arg0: !pto.ptr<f32>, %arg1: !pto.ptr<f32>, %arg2: !pto.ptr<f32>, %arg3: !pto.ptr<f32>, %arg4: i1, %arg5: i32) {
    pto.section.cube {
      %c0 = arith.constant 0 : index
      %c1 = arith.constant 1 : index
      %c128 = arith.constant 128 : index
      %c128_0 = arith.constant 128 : index
      %c128_1 = arith.constant 128 : index
      %c16384 = arith.constant 16384 : index
      %c16384_2 = arith.constant 16384 : index
      %c128_3 = arith.constant 128 : index
      %c128_4 = arith.constant 128 : index
      %0 = arith.index_cast %arg5 : i32 to index
      %1 = arith.muli %0, %c128 : index
      %2 = pto.get_block_num
      %3 = arith.index_cast %2 : i64 to index
      %4 = pto.get_block_idx
      %5 = arith.index_cast %4 : i64 to index
      %6 = arith.divsi %0, %3 : index
      %7 = arith.remsi %0, %3 : index
      %8 = arith.cmpi slt, %5, %7 : index
      %9 = arith.minui %5, %7 : index
      %10 = arith.muli %5, %6 : index
      %11 = arith.addi %10, %9 : index
      %12 = arith.select %8, %c1, %c0 : index
      %13 = arith.addi %6, %12 : index
      %14 = arith.addi %11, %13 : index
      %15 = arith.minui %14, %0 : index
      %16 = pto.make_tensor_view %arg1, shape = [%0, %c128, %c128_0] strides = [%c16384, %c128_0, %c1] : !pto.tensor_view<?x?x?xf32>
      %17 = pto.make_tensor_view %arg2, shape = [%c128_0, %c128_1] strides = [%c128_1, %c1] : !pto.tensor_view<?x?xf32>
      %18 = pto.make_tensor_view %arg0, shape = [%0, %c128, %c128_1] strides = [%c16384_2, %c128_1, %c1] : !pto.tensor_view<?x?x?xf32>
      %19 = pto.alloc_tile : !pto.tile_buf<loc=mat, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=512, pad=0>
      %20 = pto.alloc_tile : !pto.tile_buf<loc=mat, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=512, pad=0>
      %21 = pto.alloc_tile : !pto.tile_buf<loc=left, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=row_major, slayout=row_major, fractal=512, pad=0>
      %22 = pto.alloc_tile : !pto.tile_buf<loc=right, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=row_major, slayout=col_major, fractal=512, pad=0>
      %23 = pto.alloc_tile : !pto.tile_buf<loc=acc, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=1024, pad=0>
      %24 = pto.partition_view %17, offsets = [%c0, %c0], sizes = [%c128_0, %c128_4] : !pto.tensor_view<?x?xf32> -> !pto.partition_tensor_view<128x128xf32>
      pto.tload ins(%24 : !pto.partition_tensor_view<128x128xf32>) outs(%20 : !pto.tile_buf<loc=mat, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=512, pad=0>)
      pto.record_event[#pto.sync_op_type<TLOAD>, #pto.sync_op_type<TMOV_M2L>, <EVENT_ID0>]
      pto.wait_event[#pto.sync_op_type<TLOAD>, #pto.sync_op_type<TMOV_M2L>, <EVENT_ID0>]
      pto.tmov ins(%20 : !pto.tile_buf<loc=mat, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=512, pad=0>) outs(%22 : !pto.tile_buf<loc=right, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=row_major, slayout=col_major, fractal=512, pad=0>)
      scf.for %arg6 = %11 to %15 step %c1 {
        %25 = pto.partition_view %16, offsets = [%arg6, %c0, %c0], sizes = [%c1, %c128_3, %c128_0] : !pto.tensor_view<?x?x?xf32> -> !pto.partition_tensor_view<128x128xf32>
        %26 = pto.partition_view %18, offsets = [%arg6, %c0, %c0], sizes = [%c1, %c128_3, %c128_4] : !pto.tensor_view<?x?x?xf32> -> !pto.partition_tensor_view<128x128xf32>
        pto.tload ins(%25 : !pto.partition_tensor_view<128x128xf32>) outs(%19 : !pto.tile_buf<loc=mat, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=512, pad=0>)
        pto.record_event[#pto.sync_op_type<TLOAD>, #pto.sync_op_type<TMOV_M2L>, <EVENT_ID0>]
        pto.wait_event[#pto.sync_op_type<TLOAD>, #pto.sync_op_type<TMOV_M2L>, <EVENT_ID0>]
        pto.tmov ins(%19 : !pto.tile_buf<loc=mat, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=512, pad=0>) outs(%21 : !pto.tile_buf<loc=left, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=row_major, slayout=row_major, fractal=512, pad=0>)
        pto.record_event[#pto.sync_op_type<TMOV_M2L>, #pto.sync_op_type<TMATMUL>, <EVENT_ID0>]
        pto.wait_event[#pto.sync_op_type<TMOV_M2L>, #pto.sync_op_type<TMATMUL>, <EVENT_ID0>]
        pto.tmatmul ins(%21, %22 : !pto.tile_buf<loc=left, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=row_major, slayout=row_major, fractal=512, pad=0>, !pto.tile_buf<loc=right, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=row_major, slayout=col_major, fractal=512, pad=0>) outs(%23 : !pto.tile_buf<loc=acc, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=1024, pad=0>)
        pto.record_event[#pto.sync_op_type<TMATMUL>, #pto.sync_op_type<TLOAD>, <EVENT_ID0>]
        pto.wait_event[#pto.sync_op_type<TMATMUL>, #pto.sync_op_type<TLOAD>, <EVENT_ID0>]
        pto.record_event[#pto.sync_op_type<TMATMUL>, #pto.sync_op_type<TSTORE_ACC>, <EVENT_ID0>]
        pto.wait_event[#pto.sync_op_type<TMATMUL>, #pto.sync_op_type<TSTORE_ACC>, <EVENT_ID0>]
        pto.tstore ins(%23 : !pto.tile_buf<loc=acc, dtype=f32, rows=128, cols=128, v_row=128, v_col=128, blayout=col_major, slayout=row_major, fractal=1024, pad=0>) outs(%26 : !pto.partition_tensor_view<128x128xf32>)
        pto.record_event[#pto.sync_op_type<TSTORE_ACC>, #pto.sync_op_type<TMATMUL>, <EVENT_ID0>]
        pto.wait_event[#pto.sync_op_type<TSTORE_ACC>, #pto.sync_op_type<TMATMUL>, <EVENT_ID0>]
      }
    }
    return
  }
}
```

The full example (frontend + kernel launch) [is here](https://github.com/huawei-csl/pto-dsl/tree/0fe88cb6c588849bde8b6fa93ba3e7e4d2c72c97/examples/aot/matmul_dynbatch_multicore_opt), which worked correctly at ptoas commit https://github.com/zhangstevenunity/PTOAS/commit/6a8bcdbad7a0587c1d34b035617d36f1ee9c0cfa, but now gets wrong result with ptoas commit https://github.com/zhangstevenunity/PTOAS/commit/ca7bcc852ff8ca01d418d882826951d3fd1235e2

```
$ python run_matmul.py
    assert  diff <= 1e-5, diff
            ^^^^^^^^^^^^
AssertionError: tensor(19.0632, device='npu:0')
```

--- a/lib/PTO/IR/PTO.cpp
+++ b/lib/PTO/IR/PTO.cpp
@@ -4367,6 +4367,7 @@ void TMatmulOp::getEffects(SmallVectorImpl<SideEffects::EffectInstance<MemoryEff
   // Singleton -> 直接取地址
   addEffect(effects, &getLhsMutable(), MemoryEffects::Read::get());
   addEffect(effects, &getRhsMutable(), MemoryEffects::Read::get());
+  addEffect(effects, &getDstMutable(), MemoryEffects::Write::get());
 }
 
 // === TMatmulAccOp ===

